sw/airborne/modules/nav/nav_smooth.c

```diff
@@ -170,7 +170,7 @@ static void compute_ground_speed(float airspeed,
     /* g^2 -2 scal g + c = 0 */
     float scal = wind_east * cos(alpha) + wind_north * sin(alpha);
     float delta = 4 * (scal * scal - c);
-    ground_speeds[i] = scal + sqrt(delta) / 2.;
+    ground_speeds[i] = scal + sqrt(fabs(delta)) / 2.;
```

Comment:
This is not the proper way I think. The navigation routine should just fail if delta is negative.

Reply:
In the latest commit, I changed to return false on calculation failure.